### PR TITLE
Add boolean constants & basic boolean algebra

### PR DESCRIPTION
**Pull request by mathroc**
Hi!

If you don't mind contributions, here is an implementation of boolean constants & algebra. Is that what you has in mind in #2?

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -184,7 +184,14 @@ Type signatures MUST be strictly enforced by the implementation. Passing a funct
 
 * `eq` (`Any Any -> Bool`) - returns `true` if the two arguments are equal
 * `neq` (`Any Any -> Bool`) - returns `true` if the two arguments are not equal
-* `gt` (`Int Int -> Bool`) - returns `true` if the the first argument is greater than the second
-* `gteq` (`Int Int -> Bool`) - returns `true` if the the first argument is greater than or equal to the second
-* `lt` (`Int Int -> Bool`) - returns `true` if the the first argument is less than the second
-* `lteq` (`Int Int -> Bool`) - returns `true` if the the first argument is less than or equal to the second
+* `gt` (`Int Int -> Bool`) - returns `true` if the first argument is greater than the second
```
Comment (hikari-no-yume) [inline docs/spec.md:183]: oops, thanks for catching that!

File: src/lex.js [hunk 2]
```diff
@@ -114,10 +114,17 @@ window.Firth.lex = (function (Firth) {
                         break;
                     }
                 }
-                tokens.push({
-                    type: 'variable',
-                    name: token
-                });
+                if (token === 'true' || token === 'false') {
```
Comment (hikari-no-yume) [inline src/lex.js:117]: I was thinking I'd prefer to make booleans just be variables (global constants `true`, `false`) if possible. I'm not sure though. It does make sense to have them be true literals, I guess.

Comment (hikari-no-yume) [inline src/lex.js:117]: Relevant: http://python-history.blogspot.co.uk/2013/11/story-of-none-true-false.html

No harm in making them literals. We can always change it later, of course!

**Comment by hikari-no-yume** [comment]
So far as I can see this is great! Only problem: doesn't amend the spec to actually add boolean literals (and remove `true` and `false` variables, if I put them in).

**Comment by hikari-no-yume** [comment]
Oh and yeah, I'm fine with contributions, I'd really appreciate them since I don't work on this often enough.

**Comment by mathroc** [comment]
yeah I first tried to declare `true` and `false` in stdlib.js but with this implementation I had to use it like that : `false. not. show.`
By making `true` and `false` literals I can now use them without the dot : `false not. show.`

I might have missed something obvious ^^ but as you said it can be changed later

(spec amended)

**Comment by hikari-no-yume** [comment]
Oh, I mean define them in stdlib as variables, not functions. stdlib functions are just variables containing function values anyway, I just didn't make plain def yet.

**Comment by hikari-no-yume** [comment]
You only have to use the . because using a function's name merely places it on the stack, while . executes it. For a variable you need only place it on the stack, nothing to execute.

**Comment by mathroc** [comment]
Oh that was actually so simple! https://github.com/mathroc/Firth/commit/814b29e7a37d193c0cede9ab6ce1279a0efde51e ( note: I guess that would need to be flagged as a constant to disallow shadowing or changing the value )

would you like me to replace this PR with it instead ?

**Comment by hikari-no-yume** [comment]
Well, I've decided that I want to allow shadowing (and I'll close that issue) after reading Guido's blog post I linked earlier. So, I'll keep it as a literal so it can't be shadowed. We can always change it later to not be a literal, but the reverse is less doable (backwards-compatibility, though that's currently unimportant). By the way, I don't plan to allow variables to be re-assigned, so that's not a problem. (If they are to be mutable, I guess I'll require some special flag?)

**Comment by hikari-no-yume** [comment]
One final thing: you should probably move this pull request to another branch on your repository, and revert your master branch to be in sync with mine.

**Comment by hikari-no-yume** [comment]
(otherwise you're gonna have problems later!)

**Comment by hikari-no-yume** [comment]
Anyway, thanks for doing this! :)

**Pull request merged**